software/displayerang.livecodescript

```diff
@@ -0,0 +1,1402 @@
+   if mouse(mouseRight) is "down" then put "true" into tNow["3down"] -- for selection logic
+   -- (1a) location of mouse & selection indicators (since we're here)
+   put empty into tNow["inside"]
+   if pHist["3down"] is false and tNow["3down"] is true then -- reset selection when button down, otherwise retain
+      put empty into tNow["nodeSelected"]
+      put empty into tNow["linkSelected"]
+      set the visible of button "hideButt" stack "HowstrBeta" to false
+   else
+      put pHist["nodeSelected"] into tNow["nodeSelected"]
+      put pHist["linkSelected"] into tNow["linkSelected"]
+   end if
+   repeat for each word tType in "nodes links"
+      repeat for each key tNum in pGUI[tType]
+         -- find out if the mouse is inside of a graph element
+         -- within() doesn't depend on the rect, so how it treats negative space is more intuitive
+         if within(pGUI[tType][tNum]["graphic"],the mouseloc) then
+            put tNum into tNow["inside"]
+         end if
+         -- clear hilite; will be reapplied later if element still selected
+         if tType is "nodes" and tNum is not among the items of pHist["nodeSelected"] then set the showborder of pGUI[tType][tNum]["graphic"] to false
+         if tType is "links" and tNum is not among the items of pHist["linkSelected"] then  set the foregroundColor of pGUI[tType][tNum]["graphic"] to "0,220,220"
+         -- selection box
+         if tNow["3down"] and intersect(pGUI[tType][tNum]["graphic"],graphic "selectBox" stack "HowstrBeta","pixels") then
+            -- add any new selection to lists
+            if tType is "nodes" and tNum is not among the items of tNow["nodeSelected"] and tNum is not pFocus then 
+               put tNum into item (the number of items of tNow["nodeSelected"])+1 of tNow["nodeSelected"]
+               set the visible of button "hideButt" stack "HowstrBeta" to true
+            end if
+            if tType is "links" and tNum is not among the items of tNow["linkSelected"] then 
+               put tNum into item (the number of items of tNow["linkSelected"])+1 of tNow["linkSelected"]
+               set the visible of button "hideButt" stack "HowstrBeta" to true
+            end if
+         end if-- if 3down is true & intersection
+      end repeat
+   end repeat
+   -- auto-select any links attached to a selected node
+   repeat for each item tNode in tNow["nodeSelected"]
+      --      put listOfLinks(tNode) into tLinks -- ! this is a pain with the current organization, but it's short and easy to replicate
+      -- pGUI[nodes][#][abutFlow/Dive/Rise]=#,# cr #,# (link rec),(node rec)
+      repeat for each word tKind in "abutFlow abutDive abutRise"
+         repeat for each line tConnection in pGUI["nodes"][tNode][tKind]
+            put item 1 of tConnection into tCon
+            if tCon is not among the items of tNow["linkSelected"] then
+               put tCon into item (the number of items of tNow["linkSelected"])+1 of tNow["linkSelected"]
```

Comment:
@matthewmaier see my previous comment about having to count the lines concatenate when you do it this way